core/base/document.php

```diff
@@ -192,6 +193,39 @@ public static function get_editor_panel_config() {
 		];
 	}
 
+	public static function get_filtered_editor_panel_categories(): array {
+		$categories = static::get_editor_panel_categories();
+
+		foreach ( $categories as $index => $cat ) {
+			if ( isset( $cat['promotion'] ) ) {
+				$categories = self::get_panel_category_item( $cat['promotion'], $index, $categories );
+			}
+		}
+
+		return $categories;
+	}
+
+	/**
+	 * @param $promotion
+	 * @param $index
+	 * @param array $categories
+	 *
+	 * @return array
+	 */
+	private static function get_panel_category_item( $promotion, $index, array $categories ): array {
+		if ( ! Utils::has_pro() ) {
+			$categories[ $index ]['promotion'] = Filtered_Promotions_Manager::get_filtered_promotion_data(
+				$promotion,
+				'elementor/panel/' . $index . '/custom_promotion',
+				'url'
+			);
+		} else {
+			unset( $categories[ $index ]['promotion'] );
```

Comment:
Is this always set? Do we have to use `unset()`?

Reply:
@hein-obox Yes, it was easier to unset here in one place instead of building the categories array conditionally across various places.